lib/fluent/supervisor.rb

```diff
@@ -781,39 +799,33 @@ def read_config
       elsif @inline_config
         config_data << "\n" << @inline_config.gsub("\\n","\n")
       end
-      @conf = Fluent::Config.parse(config_data, config_fname, config_basedir, @use_v1_config)
+      Fluent::Config.parse(config_data, config_fname, config_basedir, @use_v1_config)
     end
 
-    def set_system_config
-      @system_config = SystemConfig.create(@conf) # @conf is set in read_config
-      @system_config.attach(self)
-      @system_config.apply(self)
+    def build_system_config(conf)
+      system_config = SystemConfig.create(conf)
+      opt = {}
+      Fluent::SystemConfig::SYSTEM_CONFIG_PARAMETERS.each do |param|
+        if @cl_opt.key?(param) && !@cl_opt[param].nil?
+          if param == :log_level && opt[:log_level] == Fluent::Log::LEVEL_INFO
+            # info level can't be specified via command line option.
+            # log_level is info here, it is default value and <system>'s log_level should be applied if exists.
+            next
+          end
+
+          opt[param] = @cl_opt[param]
+        end
+      end
+      system_config.overwrite_variables(opt)
+      system_config
     end
 
     def change_privilege
       ServerEngine::Privilege.change(@chuser, @chgroup)
     end
 
-    def init_engine(supervisor: false)
+    def init_engine
       Fluent::Engine.init(@system_config)
```

Comment:
I left it a method because others such as  Supervisor#change_privilege, Supervisor#run_configure, Supervisor#run_engine are the same situation. 
Do you think it's better to change these methods at this time? (I don't have a strong opinion about it. so if you have any opinion, I follow yours)

Reply:
I see. So it should be in other patch for refactor methods.

Reply:
i'll do it 👍